expression/builtin_time.go

```diff
@@ -1457,6 +1465,42 @@ func (b *builtinTimeSig) evalDuration(row []types.Datum) (res types.Duration, is
 	return res, isNull, errors.Trace(err)
 }
 
+type timeLiteralFunctionClass struct {
+	baseFunctionClass
+}
+
+func (c *timeLiteralFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
+	if err := c.verifyArgs(args); err != nil {
+		return nil, errors.Trace(err)
+	}
+	bf := newBaseBuiltinFuncWithTp(args, ctx, tpDuration, tpString)
+	bf.tp.Tp, bf.tp.Flen, bf.tp.Decimal = mysql.TypeDuration, 10, 0
+	sig := &builtinTimeLiteralSig{baseDurationBuiltinFunc{bf}}
+	return sig.setSelf(sig), nil
+}
+
+type builtinTimeLiteralSig struct {
+	baseDurationBuiltinFunc
+}
+
+// evalDuration evals TIME 'stringLit'.
+// See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-literals.html
+func (b *builtinTimeLiteralSig) evalDuration(row []types.Datum) (types.Duration, bool, error) {
+	sc := b.ctx.GetSessionVars().StmtCtx
+	str, isNull, err := b.args[0].EvalString(row, sc)
+	if isNull || err != nil {
+		return types.ZeroDuration, true, errors.Trace(err)
+	}
+	if !isDuration(str) {
+		return types.ZeroDuration, true, errors.Trace(types.ErrInvalidTimeFormat)
+	}
+	ret, err := types.ParseDuration(str, getFsp(str))
+	if err != nil {
+		return types.ZeroDuration, true, errors.Trace(err)
```

Comment:
no need to handle the invalid time error ?

Reply:
`DATE`, `TIME` and `TIMESTAMP` always throws errors out and never product warnings. see description for #4362